pkg/watch/health.go

```diff
@@ -85,44 +101,48 @@ func MonitorPodHealth(config *preparer.PreparerConfig, logger *logging.Logger, s
 // service it is monitoring. Every HEALTHCHECK_INTERVAL it
 // performs a health check and writes that information to
 // consul
-func (p *PodWatch) MonitorHealth(node string, store kp.Store, shutdownCh chan bool) {
+func (p *PodWatch) MonitorHealth(store kp.Store, statusChecker StatusCheck, shutdownCh chan bool) {
 	for {
 		select {
 		case <-time.After(HEALTHCHECK_INTERVAL):
-			p.checkHealth(node, p.manifest.StatusPort, store)
+			p.checkHealth(store, statusChecker)
 		case <-shutdownCh:
 			return
 		}
 	}
 }
 
-func (p *PodWatch) checkHealth(node string, port int, store kp.Store) {
-	resp, err := kp.HttpStatusCheck(node, port)
+func (p *PodWatch) checkHealth(store kp.Store, sc StatusCheck) {
+	resp, err := sc.Check()
```

Comment:
FYI: Even with a multiple-value return, Go lets you [compose function calls](http://play.golang.org/p/0xOe-dHgGU) like these together!